src/main/java/hudson/plugins/sonar/SonarGlobalConfiguration.java

```diff
@@ -72,25 +73,20 @@ public void setBuildWrapperEnabled(boolean enabled) {
   }
 
   /**
-   * Attempts to migrated data from SonarPublished, which was previously holding the global configuration.
-   * It is thread safe and will refuse to migrate if a SonarQube installation already exists in this class.
+   * Attempts to migrate data from SonarPublished, which was previously holding the global configuration.
+   * It will refuse to migrate if a SonarQube installation already exists in this class.
    * Migration will only be attempted once. 
    */
-  @Initializer(after = InitMilestone.PLUGINS_PREPARED)
-  public void migrate() {
-    if (migrated) {
+  @SuppressWarnings("deprecation")
+  @Initializer(after = InitMilestone.JOB_LOADED)
+  public void migrateData() {
+    if (dataMigrated) {
```

Comment:
The way this is done, it assumes that the method is not called concurrently. If that's not guaranteed, we could use an AtomicBoolean and do here `if (dataMigrated.getAndSet(true)) return`.

Reply:
Looking at the logstash plugin, I think we can assume the method is called only once during server startup.